value.h

```diff
@@ -164,16 +165,8 @@ static inline void truncval_to_flags(value_t *dst, match_flags flags)
 {
     assert(dst != NULL);
 
-    dst->flags.u64b &= flags.u64b;
```

Comment:
Unfortunately with bit-fields the order is undefined; Therefore you can't know what value to store in `all_flags`. Also I don't think you had put anything in `all_flags` prior to the bitwise-or op so the flags were being set incorrectly (I think it would basically be optimized into a no-op, not sure). Numeric flags should probably be in a single variable..